tests/test_training/test_opt.py

```diff
@@ -107,16 +108,16 @@ def cost_fn():
     )
     assert np.allclose(np.cos(bs.theta.value) ** 2, k / (i + k), atol=1e-2)
     assert "null_cb" in opt.callback_history
-    assert len(opt.callback_history["null_cb"]) == (len(opt.opt_history) - 1) // 3
+    assert len(opt.callback_history["null_cb"]) == (len(opt.opt_history) + 2) // 3
```

Comment:
I was expecting to have the `-1` removed, not replaced with a `+2`. There are a couple places callbacks depends on `len(opt_history)` , I'll take a look

Reply:
I think we should now have an extra `+1` on [this line](https://github.com/XanaduAI/MrMustard/blob/a4ace7c6fd838dd3c2e76d79cf0ed764bc012d37/mrmustard/training/callbacks.py#L118).

Reply:
Thanks!